django/db/models/expressions.py

```diff
@@ -470,6 +470,27 @@ def desc(self):
         return OrderBy(self, descending=True)
 
 
+class ResolvedOuterRef(F):
+    """
+    An object that contains a reference to an outer query.
+
+    In this case, the reference to the outer query has been resolved, because the
+    inner query has been used as a subquery.
+    """
+    def as_sql(self, *args, **kwargs):
+        raise FieldError(
```

Comment:
I have a test for the `ResolvedOuterRef` exception, but I'm not sure how to actually get the other one to happen. I believe it can only happen if you happened to call `OuterRef('column').as_sql(...)`, as I think `resolve_expression()` is always called first.

It may be that you can use it in a different context that causes it to not resolve first though.